lib/connection.js

```diff
@@ -377,6 +397,8 @@ class Connection extends EventEmitter {
       this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this)
     );
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
It makes no difference, in debug mode you have whole packets logged, so there is no reason to make such a switch.